// _vans@1.0.4@vans/config/karma.conf.js
// Karma configuration
// Generated on Tue Feb 07 2017 21:16:26 GMT+0800 (中国标准时间)
var path = require('path');
module.exports = function (config) {
    config.set({
        
        // base path that will be used to resolve all patterns (eg. files, exclude)
        basePath: '',
        // frameworks to use
        // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
        frameworks: ['jasmine'],
        // list of files / patterns to load in the browser
        files: [
            '../__test__/index.js'
        ],
        // list of files to exclude
        exclude: [],

        //plugins: ['karma-coverage-istanbul-reporter'],
        // preprocess matching files before serving them to the browser
        // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
        preprocessors: {
            '../__test__/index.js': 'webpack'
          },
        // test results reporter to use
        // possible values: 'dots', 'progress'
        // available reporters: https://npmjs.org/browse/keyword/karma-reporter
        reporters: ['mocha', 'coverage-istanbul'],

        coverageReporter: {
            type: 'html',
            dir: 'coverage/'
        },

        logLevel: config.LOG_DISABLE,

        // web server port
        port: 9876,
        // enable / disable colors in the output (reporters and logs)
        colors: true,
        // level of logging
        // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
        logLevel: config.LOG_INFO,
        // enable / disable watching file and executing tests whenever any file changes
        autoWatch: true,
        // start these browsers
        // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
        browsers: ['PhantomJS'],
        // Continuous Integration mode
        // if true, Karma captures browsers, runs the tests and exits
        singleRun: false,
        // Concurrency level
        // how many browser should be started simultaneous
        concurrency: Infinity,
        coverageIstanbulReporter: {
            reports: ['html', 'lcovonly', 'text-summary' ],
            fixWebpackSourcePaths: true
        },
        webpack: {
            devtool: 'inline-source-map',
            stats:'errors-only',
            module: {
                 loaders: [
                     {
                    test: /\.js$/,
                    loader:'istanbul-instrumenter-loader',
                    include: path.resolve('src/'),
                    "enforce":"post",
                    options: { esModules: false,autoWrap:true }
                },
                {
                    test: /\.js$/,
                    loader: 'babel-loader',
                    exclude: [
                        /node_modules/, 
                        ///\.index.js$/
                    ],
                   
                    query: {
                        presets: ['es2015', 'stage-0']
                    }
                }]
            }
        }
    })
}
