// _vans@1.0.4@vans/config/webpack.config.dev.js
const path = require('path');
const CleanWebpackPlugin = require('clean-webpack-plugin');
const webpack = require('webpack');
const paths = require('./paths');
module.exports = {
    entry: path.resolve(__dirname, '../src/vans.js'),

    plugins: [
        //清除发布文件夹
        new CleanWebpackPlugin(['dist'], {
            root: paths.rootPath,
            verbose: false
        }),
        //条件编译
        new webpack.DefinePlugin({
            NODE_ENV: process.env.NODE_ENV
        }),

        //Scope Hosting
        new webpack.optimize.ModuleConcatenationPlugin()
    ],

    output: {
        filename: 'vans.js',
        path: paths.distPath,
        library: 'Vans',
        libraryTarget: 'umd',
    },

    devtool: 'inline-source-map',

    module: {
        //运行ESLint
        loaders: [{
            test: /\.(js|jsx)$/,
            options: {

                eslintPath: require.resolve('eslint')

            },
            loader: 'eslint-loader',
            //include: paths.srcPath,
            enforce: 'pre'
        }, {
            test: /\.(js|jsx)$/,
            loader: 'babel-loader',
            include: paths.srcPath,
            query: {
                //presets: ['es2015', 'stage-0']
            }
        }]
    }
};
